### PR TITLE
Forms docs updates

### PR DESCRIPTION
**Pull request by kristinagore**
Please review these updates to the forms library documentation. These changes come primarily from the [React library documentation](https://react-jsonschema-form.readthedocs.io/en/latest/) and from [documentation compiled by VSA](https://github.com/department-of-veterans-affairs/va.gov-team/tree/master/teams/vsa/engineering/forms-system). Please submit review comments and questions in this PR and feel free to update the files directly. Thanks!

### CODE REVIEW

File: packages/documentation/src/pages/forms/common-patterns-for-building-forms.mdx [hunk 1]
```diff
@@ -290,5 +347,161 @@ chapterName: {
...
+    }
+  }
+}
+```
+
+In this example, `schemaform-expandUnder-indent` is applied to the `div` that surrounds `field2` and `field3`, which indents the fields. For additional styling, create a new class and add your own styles.
+
+## Creating a list loop
+
+There are times when you may need to allow the user to enter a list of items, such as a list of children or a list of spouses, and then allow them to enter detailed information about each list item on a subsequent page. This is often referred to as a `list loop`. The way this is achieved in our form system is with an `array` of form field objects. On the first page, the user enters each item in the list as the items in the `array` and then each subsequent page, where the user enters details about each list item, is then bound to the `array` in the code using a special JSON property.
```
Comment (cvalarida) [inline packages/documentation/src/pages/forms/common-patterns-for-building-forms.mdx:384]: > This is often referred to as a `list loop`.

Is this true? I've never heard this particular term in this context.

Comment (cvalarida) [inline packages/documentation/src/pages/forms/common-patterns-for-building-forms.mdx:384]:
```suggestion
There are times when you may need to allow the user to enter a list of items, such as a list of children or a list of spouses, and then allow them to enter detailed information about each list item on a subsequent page. This is often referred to as a `list loop`. The way this is achieved in our forms library is with an `array` of form field objects. On the first page, the user enters each item in the list as the items in the `array`. Each subsequent page, where the user enters details about each list item, is then bound to the `array` in the code using a special JSON property.
```

File: packages/documentation/src/pages/forms/common-patterns-for-building-forms.mdx [hunk 2]
```diff
@@ -290,5 +347,161 @@ chapterName: {
...
+    'ui:options': {
+      expandUnder: 'field1'
+    }
+  }
+}
+```
+
+In this example, `schemaform-expandUnder-indent` is applied to the `div` that surrounds `field2` and `field3`, which indents the fields. For additional styling, create a new class and add your own styles.
+
+## Creating a list loop
```
Comment (cvalarida) [inline packages/documentation/src/pages/forms/common-patterns-for-building-forms.mdx:382]: This seems like a long enough section to warrant its own page. We can link to it from here.

File: packages/documentation/src/pages/forms/common-patterns-for-building-forms.mdx [hunk 3]
```diff
@@ -290,5 +347,161 @@ chapterName: {
...
+
+### Creating an Array of Items
+
+For example, you received a design that requires a Veteran to enter in each of their chilren and then on subsequent pages to enter in details about each respective child. You need to tell the form system that you are creating an array and then do that on the first page, where you ask the user to enter in each of their children. Add the child object, which is the collection of form fields that will capture all of the initial information on the first page (in your design this is probably name, birthday, and so on), to the form schema. In this code example, the child object is named `childrenToAdd`:
```
Comment (cvalarida) [inline packages/documentation/src/pages/forms/common-patterns-for-building-forms.mdx:388]: Has this been edited?

File: packages/documentation/src/pages/forms/creating-a-form-config-file.md [hunk 4]
```diff
@@ -22,8 +21,138 @@ The form config itself is an object with many properties that determine how your
...
 
 For more information, see "[About the component hierarchy](/forms/component-hierarchy)."
+
+## How to structure your form
+
+When creating forms using the VA Forms Library it can be a little confusing figuring out how to structure your code so that it will be easy to maintain and easy to understand later. 
+
+All forms begin with a `form.js` file inside of a `/config` folder. This file is generated by the form generator and is intended to house your `schema` and your `uiSchema`. This structure works well for simple forms, but for more advanced forms we recommend that you keep this file but also break your `schema` and your `uiSchema` across smaller files that are easier to maintain. Here is a rough example of what your `form.js` will look like when you have your `schema` and `uiSchema` inside it:
```
Comment (cvalarida) [inline packages/documentation/src/pages/forms/creating-a-form-config-file.md:34]: Has this been edited? Looks like mentions of "we" were removed from the existing documentation, but not the new stuff.

File: packages/documentation/src/pages/forms/creating-a-form-config-file.md [hunk 5]
```diff
@@ -22,8 +21,138 @@ The form config itself is an object with many properties that determine how your
...
+        }
+      }
+    }
+  }
+};
+```
+
+The area you will focus most of your time on while coding your form will be the `chapters` and `pages` inside the `formConfig` object, this object stores all of the information that the VA Forms Lystem uses to "build" your form. 
+
+The most logical way to break up your code is to create a separate folder for each chapter and then inside that folder to create a folder for each page in that respective chapter with a JavaScript file named for the page, as well as an `index.js` used to import the pages.
```
Comment (cvalarida) [inline packages/documentation/src/pages/forms/creating-a-form-config-file.md:79]:
```suggestion
One way to break up your code is to create a separate folder for each chapter and then inside that folder to create a folder for each page in that respective chapter with a JavaScript file named for the page, as well as an `index.js` used to import the pages.
```

File: packages/documentation/src/pages/forms/creating-a-form-config-file.md [hunk 6]
```diff
@@ -22,8 +21,138 @@ The form config itself is an object with many properties that determine how your
...
+
+```js
+| config
+| -- chapters
+  | --- chapter1
+    |---- index.js
+    |---- page1
+      | ----- page1.js
```
Comment (cvalarida) [inline packages/documentation/src/pages/forms/creating-a-form-config-file.md:87]: Is this the recommended best practices? Seems like a lot of unnecessary nesting to me.

File: packages/documentation/src/pages/forms/form-tutorial-advanced.mdx [hunk 7]
```diff
@@ -1,19 +1,15 @@
 ---
-title: VAFS - advanced tutorial
+title: Using common definitions
```
Comment (cvalarida) [inline packages/documentation/src/pages/forms/form-tutorial-advanced.mdx:2]: This isn't what this page is about. Using common definitions is one of a few things talked about here, not the focus of the entire page.

File: packages/documentation/src/pages/forms/uischema.mdx [hunk 8]
```diff
@@ -0,0 +1,448 @@
+
+# uiSchema
```
Comment (micahchiang) [inline packages/documentation/src/pages/forms/uischema.mdx:2]: I think a lot of this is copied directly from https://react-jsonschema-form.readthedocs.io/en/latest/api-reference/uiSchema/ and there's a chance not all of it is applicable to our forked version. This should be looked at more closely to ensure only information relevant to our library version is included.

**Review by micahchiang — COMMENTED** [review]
Left comments on several things. I think if we want to consider merging any of these updates, then this PR should be split into smaller more digestible PRs. There's also some stuff in here that probably needs to be reworked.

File: packages/documentation/src/pages/forms/forms-in-production.mdx [hunk 9]
```diff
@@ -1,19 +1,13 @@
 ---
-title: Forms in production
+title: Migrating form data
 tags: updating schema, migrations
 ---
 
-# Forms in production
+# Migrating form data
```
Comment (sdk-irl) [inline packages/documentation/src/pages/forms/forms-in-production.mdx:6]: Probably s/b "Migrating forms already in production" or something parallel

File: packages/documentation/src/pages/forms/creating-custom-fields-and-widgets.md [hunk 10]
```diff
@@ -52,10 +55,318 @@ In addition to customizing fields and widgets, the VAFS code hooks into a number
...
 
 The review page also takes the config file and renders each chapter in an accordion panel. Inside a panel, the VAFS uses RJSF to render each page in a read-only view. This view uses simplified widgets and a different `FieldTemplate` component to render each form field in a definition list. The read-only view uses the RJSF `Form` component with no input elements, rendering instead with text. When a user on the review page clicks Edit for a form page, the normal widgets are used and a normal form is rendered.
 
 Each array item on a review page is rendered as read-only, and individual items can be edited independently. To accomplish this, the review `ArrayField` component renders each item in the array as its own RJSF `Form`. In addition, array fields are taken from the page's read-only view and rendered separately.
+
+## Customizing fields and widgets from RJSF
```
[the remark this answers is not in the document]
Comment (sdk-irl) [reply packages/documentation/src/pages/forms/creating-custom-fields-and-widgets.md:66]: I think, as you noted in your comment below, this page also might be copied from the RJSF docs (with zero attribution) https://react-jsonschema-form.readthedocs.io/en/latest/advanced-customization/custom-widgets-fields/

File: packages/documentation/src/pages/forms/common-variables.mdx [hunk 11]
```diff
@@ -0,0 +1,302 @@
+# Common variables
```
[the remark this answers is not in the document]
Comment (micahchiang) [reply packages/documentation/src/pages/forms/common-variables.mdx:1]: Added work to include this in [FLT ticket 117](https://github.com/department-of-veterans-affairs/formulate/issues/117)

**Comment by timwright12** [comment]
@micahchiang another old form system doc, should we close this out?

**Comment by micahchiang** [comment]
Hey @timwright12 thanks for pinging me. Yeah, feel free to close this one out. Thanks!

**Pull request closed without merging**